File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> targetCal = new HashMap<>();
        int[] result = new int[2];
        for(int i = 0; i < nums.length;i++){
            int key = target - nums[i];
            if(targetCal.containsKey(key)){
                result[0] = targetCal.get(key);
                result[1] = i;
                return result;
            }
            targetCal.put(nums[i], i);
        }
        return result;
    }
}

File: Data Structures & Algorithms/concatenation-of-array/submission-0.kt
class Solution {
    fun getConcatenation(nums: IntArray): IntArray {
        val result = IntArray(nums.size * 2)
        for(i in 0..nums.size-1){
            result[i] = nums[i]
            result[nums.size + i] = nums[i]
        }
        return result
    }
}
